### PR TITLE
Reduce View element count for border clipping scenarios

### PR DESCRIPTION
**Pull request by randy-flynn**
Use the relatively new ability to update the root element in a view to remove the 'inner panel' in scenarios where we allow the Border element to do the clipping (largely when having a borderRadius). Now the visual part of the tree will have a Border on the outside of the ViewPanel when this is needed.

As part of this change, we also don't destroy and recreate the ViewPanel when switching to use a ViewControl at the root. In fact, we will never destroy the ViewPanel now. So, more element count reduction even if it's just a temporary create.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/Microsoft/react-native-windows/pull/2327)

### CODE REVIEW

File: RNWCPP/ReactUWP/Views/ViewViewManager.cpp [hunk 1]
```diff
@@ -123,26 +131,41 @@ class ViewShadowNode : public ShadowNodeBase
...
+    return std::move(panel);
   }
 
   winrt::impl::com_ref<react::uwp::ViewControl> GetControl()
   {
-    return IsControl() ? m_view.as<react::uwp::ViewControl>() : nullptr;
+    auto control = m_view.try_as<ViewControl>();
+    auto panel = m_view.try_as<ViewPanel>();
+    auto border = m_view.try_as<winrt::Border>();
+
```
Comment (ahimberg) [inline RNWCPP/ReactUWP/Views/ViewViewManager.cpp]: these seem to be unreferenced locals?

File: RNWCPP/ReactUWP/Views/ViewViewManager.cpp [hunk 2]
```diff
@@ -200,34 +223,27 @@ XamlView ViewViewManager::CreateViewPanel(int64_t tag)
...
-  auto contentControl = contentControlPtr.as<winrt::ContentControl>();
-  contentControl.Content(newViewPanelXamlView);
 
   // TODO: Remove once existing clients stop using TouchableNativeFeedback.uwp
-  contentControl.Tapped([=](auto &&, auto &&args)
+  contentControlPtr->Tapped([=](auto &&, auto &&args)
   {
     DispatchEvent(tag, "topPress", std::move(folly::dynamic::object("target", tag)));
     args.Handled(true);
   });
```
Comment (ahimberg) [inline RNWCPP/ReactUWP/Views/ViewViewManager.cpp]: we should be able to remove this now, LPC and outlook have moved off the old touchable

File: RNWCPP/ReactUWP/Views/ViewViewManager.cpp [hunk 3]
```diff
@@ -435,75 +451,116 @@ void ViewViewManager::UpdateProperties(ShadowNodeBase* nodeToUpdate, folly::dyna
...
+    if (outerBorder.Child() == nullptr)
+      outerBorder.Child(pPanel->try_as<winrt::UIElement>());
+
+    visualRoot = outerBorder;
+  }
+
+  if (useControl)
+  {
+    // TODO NOW: auto control = pViewShadowNode->GetControl();
+    pViewShadowNode->GetControl()->Content(visualRoot);
```
Comment (ahimberg) [inline RNWCPP/ReactUWP/Views/ViewViewManager.cpp]: now?

File: RNWCPP/ReactUWP/Views/ViewViewManager.cpp (hunk 1 above)
Comment (randy-flynn) [reply RNWCPP/ReactUWP/Views/ViewViewManager.cpp]: Yeah, I'd already removed that in my branch (debug code).

File: RNWCPP/ReactUWP/Views/ViewViewManager.cpp (hunk 2 above)
Comment (randy-flynn) [reply RNWCPP/ReactUWP/Views/ViewViewManager.cpp]: I'll create a separate pull request for that then.

**Comment by randy-flynn** [comment]
> wait this needs to be against master, not rnwcpp-preview

Damn, thought I'd done that. Will recreate.

**Pull request closed without merging**